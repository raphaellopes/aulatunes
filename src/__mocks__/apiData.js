export const albums = [
  {
    'im:name': {
      label: 'Love & Hate',
    },
    'im:image': [
      {
        label: 'https://is5-ssl.mzstatic.com/image/thumb/Music124/v4/dc/1f/cb/dc1fcbe6-7ff3-7064-e0da-4ce65ee89efb/00602547859044.rgb.jpg/55x55bb.png',
        attributes: {
          height: '55',
        },
      },
      {
        label: 'https://is2-ssl.mzstatic.com/image/thumb/Music124/v4/dc/1f/cb/dc1fcbe6-7ff3-7064-e0da-4ce65ee89efb/00602547859044.rgb.jpg/60x60bb.png',
        attributes: {
          height: '60',
        },
      },
      {
        label: 'https://is3-ssl.mzstatic.com/image/thumb/Music124/v4/dc/1f/cb/dc1fcbe6-7ff3-7064-e0da-4ce65ee89efb/00602547859044.rgb.jpg/170x170bb.png',
        attributes: {
          height: '170',
        },
      },
    ],
    'im:price': {
      label: '$8.99',
      attributes: {
        amount: '8.99000',
        currency: 'USD',
      },
    },
    id: {
      label: 'https://music.apple.com/us/album/love-hate/1440832200?uo=2',
      attributes: {
        'im:id': '1440832200',
      },
    },
    'im:artist': {
      label: 'Michael Kiwanuka',
      attributes: {
        href: 'https://music.apple.com/us/artist/michael-kiwanuka/433457346?uo=2',
      },
    },
    category: {
      attributes: {
        'im:id': '15',
        term: 'R&B/Soul',
        scheme: 'https://music.apple.com/us/genre/music-r-b-soul/id15?uo=2',
        label: 'R&B/Soul',
      },
    },
  },
  {
    'im:name': {
      label: 'Cypress Grove',
    },
    'im:image': [
      {
        label: 'https://is1-ssl.mzstatic.com/image/thumb/Music123/v4/6d/61/20/6d6120c2-4eb7-8d86-14c9-5a29d26057f8/850007715281.jpg/55x55bb.png',
        attributes: {
          height: '55',
        },
      },
      {
        label: 'https://is3-ssl.mzstatic.com/image/thumb/Music123/v4/6d/61/20/6d6120c2-4eb7-8d86-14c9-5a29d26057f8/850007715281.jpg/60x60bb.png',
        attributes: {
          height: '60',
        },
      },
      {
        label: 'https://is4-ssl.mzstatic.com/image/thumb/Music123/v4/6d/61/20/6d6120c2-4eb7-8d86-14c9-5a29d26057f8/850007715281.jpg/170x170bb.png',
        attributes: {
          height: '170',
        },
      },
    ],
    'im:price': {
      label: '$9.99',
      attributes: {
        amount: '9.99000',
        currency: 'USD',
      },
    },
    id: {
      label: 'https://music.apple.com/us/album/cypress-grove/1474602159?uo=2',
      attributes: {
        'im:id': '1474602159',
      },
    },
    'im:artist': {
      label: 'Jimmy "Duck" Holmes',
      attributes: {
        href: 'https://music.apple.com/us/artist/jimmy-duck-holmes/265026843?uo=2',
      },
    },
    category: {
      attributes: {
        'im:id': '2',
        term: 'Blues',
        scheme: 'https://music.apple.com/us/genre/music-blues/id2?uo=2',
        label: 'Blues',
      },
    },
  },
];

export const songs = [
  {
    'im:name': {
      label: 'Leave The Door Open',
    },
    'im:image': [
      {
        label: 'https://is5-ssl.mzstatic.com/image/thumb/Music114/v4/36/8b/5c/368b5c12-725e-e85d-36be-5287461cc0a2/075679788337.jpg/55x55bb.png',
        attributes: {
          height: '55',
        },
      },
      {
        label: 'https://is4-ssl.mzstatic.com/image/thumb/Music114/v4/36/8b/5c/368b5c12-725e-e85d-36be-5287461cc0a2/075679788337.jpg/60x60bb.png',
        attributes: {
          height: '60',
        },
      },
      {
        label: 'https://is4-ssl.mzstatic.com/image/thumb/Music114/v4/36/8b/5c/368b5c12-725e-e85d-36be-5287461cc0a2/075679788337.jpg/170x170bb.png',
        attributes: {
          height: '170',
        },
      },
    ],
    'im:price': {
      label: '$1.29',
      attributes: {
        amount: '1.29000',
        currency: 'USD',
      },
    },
    id: {
      label: 'https://music.apple.com/us/album/leave-the-door-open/1556096966?i=1556097162&uo=2',
      attributes: {
        'im:id': '1556097162',
      },
    },
    'im:artist': {
      label: 'Bruno Mars, Anderson .Paak & Silk Sonic',
      attributes: {
        href: 'https://music.apple.com/us/artist/bruno-mars/278873078?uo=2',
      },
    },
    category: {
      attributes: {
        'im:id': '15',
        term: 'R&B/Soul',
        scheme: 'https://music.apple.com/us/genre/music-r-b-soul/id15?uo=2',
        label: 'R&B/Soul',
      },
    },
  },
  {
    'im:name': {
      label: "Don't Start Now",
    },
    'im:image': [
      {
        label: 'https://is1-ssl.mzstatic.com/image/thumb/Music124/v4/fc/aa/c9/fcaac9e1-327d-3e61-56f2-8911bc63c9d1/190295322175.jpg/55x55bb.png',
        attributes: {
          height: '55',
        },
      },
      {
        label: 'https://is5-ssl.mzstatic.com/image/thumb/Music124/v4/fc/aa/c9/fcaac9e1-327d-3e61-56f2-8911bc63c9d1/190295322175.jpg/60x60bb.png',
        attributes: {
          height: '60',
        },
      },
      {
        label: 'https://is3-ssl.mzstatic.com/image/thumb/Music124/v4/fc/aa/c9/fcaac9e1-327d-3e61-56f2-8911bc63c9d1/190295322175.jpg/170x170bb.png',
        attributes: {
          height: '170',
        },
      },
    ],
    'im:price': {
      label: '$0.69',
      attributes: {
        amount: '0.69000',
        currency: 'USD',
      },
    },
    id: {
      label: 'https://music.apple.com/us/album/dont-start-now/1484636581?i=1484636829&uo=2',
      attributes: {
        'im:id': '1484636829',
      },
    },
    'im:artist': {
      label: 'Dua Lipa',
      attributes: {
        href: 'https://music.apple.com/us/artist/dua-lipa/1031397873?uo=2',
      },
    },
    category: {
      attributes: {
        'im:id': '14',
        term: 'Pop',
        scheme: 'https://music.apple.com/us/genre/music-pop/id14?uo=2',
        label: 'Pop',
      },
    },
  },
];
