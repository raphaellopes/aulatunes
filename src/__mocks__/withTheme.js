import React from 'react';
import { ThemeProvider } from 'styled-components';

export const withTheme = (Component) => (
  <ThemeProvider theme={{ mode: 'light' }}>
    {Component}
  </ThemeProvider>
);
