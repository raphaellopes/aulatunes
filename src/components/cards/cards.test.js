import { render } from '@testing-library/react';

import { withTheme } from '../../__mocks__/withTheme';
import {
  Card, CardContent, CardImage, CardTitle, CardSubtitle, CardLabel,
  CardRow, CardList,
} from './index';
import {
  CardPlaceholder, CardImagePlaceholder, CardTitlePlaceholder, CardSubtitlePlaceholder,
} from './placeholder';

describe('components | cards', () => {
  const renderComponent = (Component, label) => {
    test(`should render the ${label}`, () => {
      const { container } = render(withTheme(Component));
      expect(container).toMatchSnapshot();
    });
  };

  renderComponent(<Card>Some text</Card>, 'Card');
  renderComponent(<Card variant="secondary">Some text</Card>, 'Card with secondary variant');
  renderComponent(<Card onClick={jest.fn}>Some text</Card>, 'Card with cursor');
  renderComponent(<CardContent>Some text</CardContent>, 'CardContent');
  renderComponent(<CardImage src="http://some-image" alt="some text" />, 'CardImage');
  renderComponent(<CardTitle>Some text</CardTitle>, 'CardTitle');
  renderComponent(<CardSubtitle>Some text</CardSubtitle>, 'CardSubtitle');
  renderComponent(<CardLabel>Some text</CardLabel>, 'CardLabel');
  renderComponent(<CardRow>Some text</CardRow>, 'CardRow');
  renderComponent(<CardList>Some text</CardList>, 'CardList');
  renderComponent(<CardPlaceholder />, 'CardPlaceholder');
  renderComponent(<CardImagePlaceholder />, 'CardImagePlaceholder');
  renderComponent(<CardTitlePlaceholder />, 'CardTitlePlaceholder');
  renderComponent(<CardSubtitlePlaceholder />, 'CardSubtitlePlaceholder');
});
