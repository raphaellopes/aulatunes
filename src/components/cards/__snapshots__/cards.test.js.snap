// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | cards should render the Card 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 10px;
  border-radius: 10px;
  box-shadow: 0 1px 4px 0 rgba(49,64,92,.1);
  background-color: #ffffff;
}

<div>
  <div
    class="c0"
  >
    Some text
  </div>
</div>
`;

exports[`components | cards should render the Card with cursor 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 10px;
  border-radius: 10px;
  box-shadow: 0 1px 4px 0 rgba(49,64,92,.1);
  background-color: #ffffff;
  cursor: pointer;
}

<div>
  <div
    class="c0"
  >
    Some text
  </div>
</div>
`;

exports[`components | cards should render the Card with secondary variant 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 10px;
  border-radius: 10px;
  box-shadow: 0 1px 4px 0 rgba(49,64,92,.1);
  background-color: #00E0B7;
}

.c0 p {
  color: #282b31;
}

<div>
  <div
    class="c0"
  >
    Some text
  </div>
</div>
`;

exports[`components | cards should render the CardContent 1`] = `
.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div>
  <div
    class="c0"
  >
    Some text
  </div>
</div>
`;

exports[`components | cards should render the CardImage 1`] = `
.c0 {
  margin-right: 10px;
  border-radius: 5px;
  width: 60px;
  height: 60px;
}

@media (min-width:768px) {
  .c0 {
    width: 170px;
    height: 170px;
  }
}

<div>
  <img
    alt="some text"
    class="c0"
    src="http://some-image"
  />
</div>
`;

exports[`components | cards should render the CardImagePlaceholder 1`] = `
.c0 {
  display: inline-block;
  background-color: #d6d6d7;
  border-radius: 5px;
}

.c1 {
  margin-right: 10px;
  width: 60px;
  height: 60px;
}

@media (min-width:768px) {
  .c1 {
    width: 170px;
    height: 170px;
  }
}

<div>
  <span
    class="c0 c1"
  />
</div>
`;

exports[`components | cards should render the CardLabel 1`] = `
.c0 {
  font-family: sans-serif;
  font-weight: 400;
}

.c1 {
  font-size: 14px;
}

.c2 {
  color: #bdbcbe;
}

<div>
  <p
    class="c0 c1 c2"
  >
    Some text
  </p>
</div>
`;

exports[`components | cards should render the CardList 1`] = `
.c0 {
  margin-top: 16px;
}

<div>
  <div
    class="c0"
  >
    Some text
  </div>
</div>
`;

exports[`components | cards should render the CardPlaceholder 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 10px;
  border-radius: 10px;
  box-shadow: 0 1px 4px 0 rgba(49,64,92,.1);
  background-color: #ffffff;
}

.c3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c1 {
  display: inline-block;
  background-color: #d6d6d7;
  border-radius: 5px;
}

.c2 {
  margin-right: 10px;
  width: 60px;
  height: 60px;
}

.c4 {
  height: 26px;
  width: 90%;
}

.c5 {
  height: 16px;
  width: 70%;
}

@media (min-width:768px) {
  .c2 {
    width: 170px;
    height: 170px;
  }
}

<div>
  <div
    class="c0"
  >
    <span
      class="c1 c2"
    />
    <div
      class="c3"
    >
      <span
        class="c1 c4"
      />
      <span
        class="c1 c5"
      />
    </div>
  </div>
</div>
`;

exports[`components | cards should render the CardRow 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0:not(:last-child) {
  margin-bottom: 10px;
}

.c0 .sc-gKsewC:not(:last-child) {
  margin-bottom: 10px;
}

@media (min-width:768px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }

  .c0 .sc-gKsewC {
    -webkit-flex-basis: 32.6%;
    -ms-flex-preferred-size: 32.6%;
    flex-basis: 32.6%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    margin-bottom: 0;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }

  .c0 .sc-gKsewC > img,
  .c0 .sc-gKsewC > span {
    margin: 16px auto;
    -webkit-align-self: center;
    -ms-flex-item-align: center;
    align-self: center;
  }

  .c0 .sc-gKsewC:not(:last-child) {
    margin-right: 10px;
    margin-bottom: 0;
  }
}

<div>
  <div
    class="c0"
  >
    Some text
  </div>
</div>
`;

exports[`components | cards should render the CardSubtitle 1`] = `
.c0 {
  font-family: sans-serif;
  font-weight: 400;
}

.c1 {
  font-size: 16px;
}

.c2 {
  color: #bdbcbe;
  margin-bottom: 5px;
}

<div>
  <p
    class="c0 c1 c2"
  >
    Some text
  </p>
</div>
`;

exports[`components | cards should render the CardSubtitlePlaceholder 1`] = `
.c0 {
  display: inline-block;
  background-color: #d6d6d7;
  border-radius: 5px;
}

.c1 {
  height: 16px;
  width: 70%;
}

<div>
  <span
    class="c0 c1"
  />
</div>
`;

exports[`components | cards should render the CardTitle 1`] = `
.c0 {
  font-family: sans-serif;
  font-weight: 600;
}

.c1 {
  font-size: 22px;
}

.c2 {
  color: #31405c;
}

<div>
  <h1
    class="c0 c1 c2"
  >
    Some text
  </h1>
</div>
`;

exports[`components | cards should render the CardTitlePlaceholder 1`] = `
.c0 {
  display: inline-block;
  background-color: #d6d6d7;
  border-radius: 5px;
}

.c1 {
  height: 26px;
  width: 90%;
}

<div>
  <span
    class="c0 c1"
  />
</div>
`;
